bcipy/simulator/ui/gui.py

```diff
@@ -0,0 +1,679 @@
+    """Recursively build a Tree with directory contents.
+    Adapted from rich library examples/tree.py
+
+    Parameters
+    ----------
+        directory - root directory of the tree
+        tree - Tree object that gets recursively updated
+        skip_dir - predicate to determine if a directory should be skipped
+    """
+    # Sort dirs first then by filename
+    paths = sorted(
+        Path(directory).iterdir(),
+        key=lambda path: (path.is_file(), path.name.lower()),
+    )
+    for path in paths:
+        # Remove hidden files
+        if path.name.startswith(".") or path.is_file() or (path.is_dir()
+                                                           and skip_dir(path)):
+            continue
+        if path.is_dir():
+            branch = QTreeWidgetItem(tree, [path.name])
+            walk_directory(path, branch, skip_dir)
+
+
+def is_ancestor(directory: Path, paths: List[Path]) -> bool:
+    """Check if the given directory is the ancestor of any of the provided paths."""
+    for path in paths:
+        if directory in path.parents:
+            return True
+    return False
+
+
+class DirectoryTree(QWidget):
+    """Display a tree of directories"""
+
+    def __init__(self,
+                 parent: Optional[QWidget] = None,
+                 parent_directory: Optional[str] = None,
+                 selected_subdirectories: Optional[List[str]] = None):
+        super().__init__(parent=parent)
+        self.parent_directory = parent_directory
+        self.paths = selected_subdirectories or []
+
+        self.label = "Selected Directories"
+        self.tree = self.make_tree()
+        self.layout = QVBoxLayout()
+        self.layout.addWidget(
+            LabeledWidget(self.label, self.tree, label_size=12))
+        self.layout.addWidget(self.tree)
+        self.setLayout(self.layout)
+
+    def update(self, parent_directory: str,
+               selected_subdirectories: List[str]) -> None:
+        """Update the widget."""
+        self.parent_directory = parent_directory
+        self.paths = selected_subdirectories or []
+
+        clear_layout(self.layout)
+        if self.parent_directory:
+            self.tree = self.make_tree()
+            self.layout.addWidget(
+                LabeledWidget(self.label, self.tree, label_size=12))
+
+    def make_tree(self) -> QTreeWidget:
+        """Initialize the tree widget"""
+
+        tree = QTreeWidget()
+        if self.parent_directory:
+            tree.setColumnCount(1)
+            tree.setHeaderLabels([self.parent_directory])
+            walk_directory(Path(self.parent_directory),
+                           tree,
+                           skip_dir=self.skip_path)
+        return tree
+
+    def skip_path(self, directory: Path) -> bool:
+        """Should the given directory be skipped"""
+        if directory in self.paths or is_ancestor(directory, self.paths):
+            return False
+        return True
+
+
+class ChooseFileInput(QWidget):
```

Comment:
We have similar code in the `GUI` module for getting files and directories. Could we leverage that here? We will be collapsing some of the GUI code (and likely passing around a single QApplication Instance) to package in the future, which is why I ask.

Reply:
These widgets are a bit different in that they represent just the selection input and not the input, label, checkbox, and separator, all of which would affect the layout. These also have hooks for a change action, which is important for this form, but not the parameters GUI.

I do think our existing widgets could be updated to use these components as the underlying input, which would eliminate the duplicate code between the two. That would be a good refinement and when we do that work these could move to a common location.